smoketest/mqtt/mqtt.go

```diff
@@ -0,0 +1,135 @@
+package mqtt
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+
+	"github.com/eclipse/paho.golang/paho"
+)
+
+// TopicData is a tuple for Topic and Message.
+type TopicData struct {
+	Topic   string
+	Message []byte
+}
+
+// MQTT is connection type for
+type MQTT struct {
+	client            *paho.Client
+	connectProperties *paho.Connect
+	qos               byte
+	in                chan *TopicData // Is used to send respons messages of a handler downwards
+}
+
+func (m MQTT) In() <-chan *TopicData {
+	return m.in
+}
+
+func (m MQTT) Close() error {
+	close(m.in)
+	return m.client.Disconnect(&paho.Disconnect{ReasonCode: 0})
+}
+
+func (m MQTT) register(topic string) error {
+
+	m.client.Router.RegisterHandler(topic, func(p *paho.Publish) {
+		m.in <- &TopicData{Topic: topic, Message: p.Payload}
```

Comment:
Maybe I am a bit confused, but I think the name `in` is very irritating, as it is used to receive the actual data send via mqtt (if I understand it correctly)

Reply:
It is the data received not send from mqtt hence in for incoming.